fuels-contract/src/contract.rs

```diff
@@ -159,7 +148,35 @@ impl Contract {
         if let Some(e) = encoded_args {
             script_data.extend(e)
         }
+        Ok((script, script_data))
+    }
 
+    /// Calls an already-deployed contract code.
+    /// Note that this is a "generic" call to a contract
+    /// and it doesn't, yet, call a specific ABI function in that contract.
+    /// We need a wallet to pay for the transaction fees (even though they are 0 right now)
```

Comment:
I think this is a very confusing comment. `"generic"` here is very vague.

> it doesn't, yet, call a specific ABI function in that contract

This isn't correct, is it? It does call a specific ABI function in the contract. The function is `encoded_selector` with `encoded_args` as arguments. And it is called at the end, through `script.call(fuel_client).await` (or `.simulate`).

> We need a wallet to pay for the transaction fees

And we do have a wallet in here: `wallet: LocalWallet`.

Reply:
I think that's just copy-paste of the original comment, which as you say is outdated.

Reply:
The wallet bit was just to justify that it's there even though transactions are free